### PR TITLE
Better sorting of the modes table

### PR DESCRIPTION
**Pull request by cquiroz**
With the update of react table of 8.9.2 the logic of sorting undefined changed a bit.
After some experiments I thought it maybe be better to take full control defining different orderings for each direction in code.
Now the times are sorted such that all the non available values go to the end regardless if we are ascending or descending.
There are also some small ui touches for the configuration tile

### CODE REVIEW

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 1]
```diff
@@ -499,37 +508,41 @@ private object SpectroscopyModesTable extends TableHooks:
...
           (configuration, requirements, sortedRows) =>
             selectedIndex.setState(
               selectedRowIndex(configuration, requirements.wavelength, sortedRows)
             )
       }
       // Set the selected config if the rows change because it might have different itc data.
       // Note, we use rows for the dependency, not sorted rows, because sorted rows also changes with sort.
-      .useEffectWithDepsBy((_, _, _, rows, _, _, _, _, _, _, _) => rows) {
-        (props, _, _, _, _, _, _, _, _, sortedRows, selectedIndex) => _ =>
+      .useEffectWithDepsBy((_, _, _, rows, _, _, timeSort, _, _, _, _, _) => rows) {
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:544]: Oh right, I had to do some experiments trying to find a bug on saving table prefrences

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 2]
```diff
@@ -499,37 +508,41 @@ private object SpectroscopyModesTable extends TableHooks:
...
+          table.setSorting(s) *>
+            sorting.setState(s.value.headOption) *> scrollTo
+              .setState(ScrollTo.Scroll)
       }
-      .useMemoBy((_, _, _, rows, _, _, _, table, _) => (rows, table.getState().sorting)) {
-        (_, _, _, _, _, _, _, table, _) => (_, _) =>
-          table.getSortedRowModel().rows.map(_.original).toList
+      .useMemoBy((_, _, _, rows, _, _, timeSort, _, table, _) =>
+        (rows, timeSort.value, table.getState().sorting)
+      ) { (_, _, _, _, _, _, sorting, _, table, _) => (_, s, _) =>
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:519]: It seems named `s` is not used in the hook body.

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 3]
```diff
@@ -499,37 +508,41 @@ private object SpectroscopyModesTable extends TableHooks:
       // By having it as state with the following `useEffectWithDepsBy`, the scrollTo effect will run
       // in the following "hook cycle" and get the proper index.
       .useState(ScrollTo.Scroll)
-      .useEffectWithDepsBy((_, _, _, _, _, _, _, table, _) => table.getState().sorting) {
-        (_, _, _, _, _, _, _, _, scrollTo) => _ => scrollTo.setState(ScrollTo.Scroll)
+      .useEffectWithDepsBy((_, _, _, _, _, _, _, _, table, _) => table.getState().sorting) {
+        (_, _, _, _, _, _, sorting, _, table, scrollTo) => s =>
+          table.setSorting(s) *>
+            sorting.setState(s.value.headOption) *> scrollTo
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:514]: I'm not sure the whole mechanism will work if time is not the first column in the sort. Eg: we can sort first by instrument and then by time.

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala [hunk 4]
```diff
@@ -208,7 +202,17 @@ private object SpectroscopyModesTable extends TableHooks:
   private given Order[InstrumentRow#FPU]     = Order.by(_.toString)
   private given Order[BasicConfigAndItc]     = Order.by(_.configuration.configurationSummary)
 
-  private given Order[TimeSpan | Unit] = Order.by(_.toOption)
+  private def timeOrder(sort: Option[ColumnSort]): Order[Option[TimeSpan]] =
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:205]: Does this mechanism push undefined values to the bottom when the column is not sorted?

Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:205]: Nope, the order is defined by the other columns

File: explore/src/main/scala/explore/config/SpectroscopyModesTable.scala (hunk 3 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/config/SpectroscopyModesTable.scala:514]: II don't think multi sort is enabled in this table. AFAIK only one column is sorted

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**